assert/assertions_test.go

```diff
@@ -2807,12 +2807,31 @@ func TestNeverTrue(t *testing.T) {
 	False(t, Never(mockT, condition, 100*time.Millisecond, 20*time.Millisecond))
 }
 
-func TestEventuallyIssue805(t *testing.T) {
+// Check that a long running condition doesn't block Eventually.
+// See issue 805 (and its long tail of following issues)
+func TestEventuallyTimeout(t *testing.T) {
 	mockT := new(testing.T)
 
 	NotPanics(t, func() {
-		condition := func() bool { <-time.After(time.Millisecond); return true }
+		t.Log("start")
```

Comment:
Good spot on the t.Log @stevenh. Looking at the rest of the package there is no other test which logs on success, so we should probably hold that convention.

Reply:
Well, I think that the logs are helpful to show what happens. In fact the lack of logging was what made debugging the issue harder.

The sequence of events is really important in this test, and the log output allows to validate visually that everything happens in the right order. I think this will be useful for regression testing if the test has to be changed later.

This is the only test that has verbose log output, but I think that's is what verbose output is for. This is a case that is worth it.

The log messages could be improved though. Suggestions welcome.

Reply:
In my opinion good comments would be more valuable, you already have some.

The goal as you say is to help those who might have to debug it in the future, by making it clear why flow is how it is.

I don't think you need logging to achieve this, and the current output e.g. `ok1` and `ok2` doesn't add any value as it doesn't describe what is happening or why, leaving the user thinking what is `ok1` etc?

Reply:
Logs removed.